spec/helpers/tcp_gate_scheduler.rb

```diff
@@ -107,15 +105,19 @@ def write(amount=5)
 							read_str = @internal_io.read_nonblock(len)
 							print_data("write fd:#{@internal_io.fileno}->#{@external_io.fileno}", read_str)
 							@external_io.write(read_str)
+							if until_writeable
+								res = IO.select(nil, [until_writeable], nil, 0)
```

Comment:
`IO.select` is not compatible with the fiber scheduler FYI.

Reply:
Thanks, but the timeout is zero here, so that it's always nonblocking. The mechanism is so, that in the write direction only as much data is transferred as necessary to make the observed IO writable again. This should trigger any blocking issues while writing at best.

Reply:
You could just use `until_writeable.wait_writable(0)`?